### PR TITLE
Add `ThemelessFluentProvider` implementation

### PR DESCRIPTION
**Pull request by spmonahan**
Implements `ThemelessFluentProvider` component per [FluentUI#30241](https://github.com/microsoft/fluentui/issues/30241).

Adds:

1. Implementation
2. Documentation/Storybook example
3. Tests

Depends on https://github.com/microsoft/fluentui-contrib/pull/153

### CODE REVIEW

**Review by tudorpopams — APPROVED** [review]

File: packages/react-themeless-provider/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export function App() {
+  React.useEffect(() => {
+    // Example of how to create a style that will piece the shadow DOM.
+    // This does not need to be created by React and can be created
+    // by a host Web Component application for example.
+    const cssRule = createCSSRuleFromTheme(':root', webLightTheme);
+    const stylesheet = new CSSStyleSheet();
+    stylesheet.insertRule(cssRule);
+
+    document.adoptedStyleSheets = [...document.adoptedStyleSheets, stylesheet];
```
Comment (layershifter) [inline packages/react-themeless-provider/README.md:32]: Let's add a cleanup to ensure that sheet gets removed once a component is unmounted.

Comment (layershifter) [inline packages/react-themeless-provider/README.md:32]: This example is a bit confusing. We insert there to a host document there, right? Can we insert using regular `style` tag?

File: packages/react-themeless-provider/src/components/ThemelessFluentProvider/ThemelessFluentProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  if (a) {
+    return a;
+  }
+
+  return b;
+}
+
+function useThemelessFluentProviderState(
+  props: ThemelessFluentProviderProps,
+  ref: React.Ref<HTMLElement>
```
Comment (layershifter) [inline packages/react-themeless-provider/src/components/ThemelessFluentProvider/ThemelessFluentProvider.tsx:43]:
```suggestion
  ref: React.Ref<HTMLDivElement>
```

File: packages/react-themeless-provider/src/components/ThemelessFluentProvider/ThemelessFluentProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      getIntrinsicElementProps('div', {
+        ...props,
+        dir,
+        // FIXME:
+        // `ref` is wrongly assigned to be `HTMLElement` instead of `HTMLDivElement`
+        // but since it would be a breaking change to fix it, we are casting ref to it's proper type
+        ref: useMergedRefs(
+          ref,
+          useFocusVisible<HTMLDivElement>({ targetDocument })
+        ) as React.Ref<HTMLDivElement>,
```
Comment (layershifter) [inline packages/react-themeless-provider/src/components/ThemelessFluentProvider/ThemelessFluentProvider.tsx:85]:
```suggestion
        ref: useMergedRefs(
          ref,
          useFocusVisible<HTMLDivElement>({ targetDocument })
        ),
```

You should be able to remove this FIXME block this proper `ref` typing

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -8,10 +8,14 @@ packages/react-data-grid-react-window-grid/ @microsoft/ms-fabric
...
 packages/pierce-dom/ @spmonahan
 packages/houdini-utils @microsoft/teams-prg
+<<<<<<< HEAD
 packages/react-headless-provider/ @microsoft/teams-prg
 packages/react-resize-handler/ @microsoft/teams-prg
 packages/stylelint-plugin/ @microsoft/teams-prg
 packages/react-themeless-provider/ @spmonahan
+=======
+packages/react-themeless-provider @spmonahan
+>>>>>>> chore: scaffold react-themeless-provider package
```
Comment (layershifter) [inline .github/CODEOWNERS:18]: Bad merge?

File: packages/react-themeless-provider/README.md [hunk 5]
```diff
@@ -1,11 +1,43 @@
...
+export function App() {
+  React.useEffect(() => {
+    // Example of how to create a style that will piece the shadow DOM.
+    // This does not need to be created by React and can be created
+    // by a host Web Component application for example.
+    const cssRule = createCSSRuleFromTheme(':root', webLightTheme);
+    const style = document.createElement('style');
+    document.head.appendChild(style);
+    style.sheet.insertRule(cssRule);
+  }, []);
```
Comment (layershifter) [inline packages/react-themeless-provider/README.md:32]:
```suggestion
    return () => {
      document.head.appendChild(style);
    }
  }, []);
```

File: packages/react-themeless-provider/src/utils/createCSSStyleSheetFromTheme.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { createCSSRuleFromTheme } from '@fluentui/react-components';
+import type { PartialTheme } from '@fluentui/react-theme';
```
Comment (layershifter) [inline packages/react-themeless-provider/src/utils/createCSSStyleSheetFromTheme.ts:2]:
```suggestion
import type { PartialTheme } from '@fluentui/react-components';
```

this is exported from `@fluentui/react-components`

File: packages/react-themeless-provider/package.json [hunk 7]
```diff
@@ -6,12 +6,18 @@
...
+  "devDependencies": {
+    "@fluentui/react-jsx-runtime": "^9.0.32",
+    "@fluentui-contrib/react-shadow": "^0.1.0"
+  },
   "peerDependencies": {
     "@fluentui/react-components": ">=9.46.3 <10.0.0",
+    "@fluentui/react-icons": ">=2.0.224 <3.0.0",
+    "@fluentui/react-jsx-runtime": ">=9.0.32 <10.0.0",
+    "@fluentui/react-shared-contexts": ">=9.7.2 <10.0.0",
+    "@fluentui/react-theme": ">=9.1.16 <10.0.0",
```
Comment (layershifter) [inline packages/react-themeless-provider/package.json:18]:
```suggestion
```

File: packages/react-themeless-provider/package.json [hunk 8]
```diff
@@ -6,12 +6,18 @@
...
   "typings": "./src/index.d.ts",
-  "private": true,
+  "devDependencies": {
+    "@fluentui/react-jsx-runtime": "^9.0.32",
+    "@fluentui-contrib/react-shadow": "^0.1.0"
+  },
   "peerDependencies": {
     "@fluentui/react-components": ">=9.46.3 <10.0.0",
+    "@fluentui/react-icons": ">=2.0.224 <3.0.0",
+    "@fluentui/react-jsx-runtime": ">=9.0.32 <10.0.0",
```
Comment (layershifter) [inline packages/react-themeless-provider/package.json:16]: Let's have this as a regular dep

File: packages/react-themeless-provider/package.json [hunk 9]
```diff
@@ -6,12 +6,18 @@
...
+    "@fluentui/react-shared-contexts": ">=9.7.2 <10.0.0",
+    "@fluentui/react-theme": ">=9.1.16 <10.0.0",
+    "@griffel/react": ">=1.5.14 <2.0.0",
```
Comment (layershifter) [inline packages/react-themeless-provider/package.json:19]: We should probably also put `@griffel/react` as a regular dep.

I don't think that it's a good idea to have anything else except `@fluentui/react-components` as `peerDependency` as it's a single package that we expect to be installed by consumers.

**Review by layershifter — APPROVED** [review]

File: .github/CODEOWNERS (hunk 4 above)
Comment (spmonahan) [reply .github/CODEOWNERS:18]: Yep :(

**Pull request merged**